maas/resource_maas_boot_resource.go

```diff
@@ -0,0 +1,174 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceMAASBootResources() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a resource to manage MAAS bootresources.",
+		CreateContext: resourceBootResourcesCreate,
+		ReadContext:   resourceBootResourcesRead,
+		UpdateContext: resourceBootResourcesUpdate,
+		DeleteContext: resourceBootResourcesDelete,
+
+		Schema: map[string]*schema.Schema{
+			"boot_source": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The boot source database ID this resource is associated with.",
+			},
+			"boot_source_selections": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "The set of database IDs for boot source selections to attach to this boot resource",
+				Elem:        &schema.Schema{Type: schema.TypeInt},
+			},
+		},
+	}
+}
+
+func resourceBootResourcesCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ClientConfig).Client
+
+	for isImporting(client) {
+		time.Sleep(5 * time.Second)
+	}
+
+	resources, err := getBootResources(client, "synced")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	resourceMap := make(map[string]struct{})
+	for _, res := range resources {
+		resourceMap[res.Name] = struct{}{}
+	}
+
+	bootselections := d.Get("boot_source_selections").(*schema.Set).List()
+	if len(bootselections) == 0 {
+		return diag.Errorf("At least one boot source selection must be added to the boot resources")
+	}
+
+	for _, bootselection := range bootselections {
+		bootselection, err := getBootSourceSelection(client, d.Get("boot_source").(int), bootselection.(int))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		// check the selection has it's resource created
+		if _, exists := resourceMap[fmt.Sprintf("%s/%s", bootselection.OS, bootselection.Release)]; !exists {
+			return diag.Errorf("Boot Resource missing for %s/%s", bootselection.OS, bootselection.Release)
```

Comment:
Should we re-trigger import and wait? Maybe not.

Reply:
I think it's enough to tell the user there was an issue, and they can modify the plan and trigger an import